fbw-a32nx/src/systems/systems-host/systems/FWC/PseudoFWC.ts

```diff
@@ -1332,6 +1503,194 @@ export class PseudoFWC {
       (onGroundCount > 1 && raInvalid);
     this.aircraftOnGround.set(this.onGroundConf.write(this.onGroundImmediate, deltaTime));
 
+    // AP OFF
+    const apEngaged = SimVar.GetSimVarValue('L:A32NX_AUTOPILOT_ACTIVE', 'Bool');
+    this.autoPilotDisengagedInstantPulse.write(apEngaged, deltaTime);
+
+    const apDiscPressedInLast1p8SecBeforeThisCycle = this.autoPilotInstinctiveDiscPressedInLast1p9Sec.read();
+    this.autoPilotInstinctiveDiscPressedInLast1p9Sec.write(this.autoPilotInstinctiveDiscPressedPulse.read(), deltaTime);
+
+    const voluntaryApDisc =
+      this.autoPilotDisengagedInstantPulse.read() && this.autoPilotInstinctiveDiscPressedInLast1p9Sec.read();
+    this.autoPilotOffVoluntaryEndAfter1p9s.write(voluntaryApDisc, deltaTime);
+    this.autoPilotOffVoluntaryDiscPulse.write(voluntaryApDisc, deltaTime);
+
+    this.autoPilotOffVoluntaryFirstCavalryChargeActive.write(this.autoPilotOffVoluntaryDiscPulse.read(), deltaTime);
+
+    this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.write(
+      this.autoPilotInstinctiveDiscPressedPulse.read() &&
+        (this.autoPilotInstinctiveDiscCountSinceLastFwsCycle > 1 || apDiscPressedInLast1p8SecBeforeThisCycle),
+      deltaTime,
+    );
+
+    this.autoPilotOffVoluntaryMemory.write(
+      this.autoPilotOffVoluntaryDiscPulse.read(),
+      apEngaged ||
+        this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.read() ||
+        !this.autoPilotOffVoluntaryEndAfter1p9s.read(),
+    );
+
+    const discPbPressedAfterDisconnection =
+      !this.autoPilotDisengagedInstantPulse.read() &&
+      (this.autoPilotInstinctiveDiscPressedPulse.read() || masterWarningButtonLeft || masterWarningButtonRight);
+
+    this.autoPilotOffUnacknowledged.write(
+      this.autoPilotDisengagedInstantPulse.read(),
+      apEngaged || this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.read() || discPbPressedAfterDisconnection,
+    );
+
+    this.autoPilotOffInvoluntaryMemory.write(
+      !apEngaged && !this.autoPilotOffVoluntaryMemory.read() && this.autoPilotOffUnacknowledged.read(),
+      !this.autoPilotOffUnacknowledged.read(),
+    );
+    this.autoPilotOffInvoluntary.set(this.autoPilotOffInvoluntaryMemory.read());
+    this.autoPilotOffShowMemo.set(this.autoPilotOffVoluntaryMemory.read() || this.autoPilotOffInvoluntaryMemory.read());
+
+    if (this.autoPilotDisengagedInstantPulse.read()) {
+      // Request quiet CRC one time
+      this.requestMasterWarningFromApOff = true;
+      this.soundManager.setVolume(FwsAuralVolume.Attenuated);
```

Comment:
It should always be full volume, the only difference on the A320 for FWS volume is engines running vs engines off.  The minimum cavalry charges played are also quite different but that can be left for future work I guess as its out of scope of the intent of the PR  😄  .

Reply:
Interesting! Then I'll remove that part. I believe this setting is without effect anyways at the moment. About the minimum number of cav charges: I implemented that differently for a32nx and a380x: a380x CC can be cancelled instantly, a32nx plays a sequence of three CCs I believe (That's the "cavalryChargeOnce" event, one sound file with a number of CCs). Have you tried it in the sim?